src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java

```diff
@@ -13,14 +13,19 @@
 /**
  * Interface for the workflow setup of different building blocks.
  */
-public interface Workflow {
+public interface WorkflowStep {
 
     /**
      * Triggers the processing of the building block.
      *
      * @return CompletableFuture of the building block.
-     * @throws Exception if execution fails
      */
-    CompletableFuture<Workflow> execute() throws Exception;
+    CompletableFuture<WorkflowStep> execute();
```

Comment:
Let's make this signature
```java
CompletableFuture<WorkflowData> execute(WorkflowData data);
```
and then create a [tagging interface](https://www.tutorialspoint.com/What-is-a-marker-or-tagged-interface-in-Java) `WorkflowData` that we can add functionality to later.

Reply:
My gradle check took 5 mins. Pushed the change.